From ca07d42ca64271689f1a1ea2219b4909bd97f7c1 Mon Sep 17 00:00:00 2001
From: Boudewijn Rempt <boud@valdyas.org>
Date: Wed, 2 Oct 2019 12:48:28 +0200
Subject: [PATCH] Work around a crash in Qt when converting a QDomDocument to a
 string

In Qt 5.14, if that is done in a thread which is not the gui thread,
QRegularExpression won't have a QThreadStorage and will crash. It is
weird, though, that Qt's exit handlers clear static variables in
non-gui threads.

I'm not sure what changed in Qt: the exit handler or QRegularExpression.

Since we save the tags on every change, the saving on exit isn't strictly
needed, so we can remove that to work around this issue in Qt.

BUG:410867
---
 libs/widgets/KoResourceTagStore.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libs/widgets/KoResourceTagStore.cpp b/libs/widgets/KoResourceTagStore.cpp
index b9ea9558bb..34566c211c 100644
--- a/libs/widgets/KoResourceTagStore.cpp
+++ b/libs/widgets/KoResourceTagStore.cpp
@@ -28,6 +28,9 @@
 #include <KoResourcePaths.h>
 #include <KoResourceServer.h>
 
+#include <QApplication>
+#include <QThread>
+
 #include <kis_debug.h>
 
 #define BLACKLISTED "blacklisted" ///< xml tag for blacklisted tags
@@ -55,7 +58,6 @@ KoResourceTagStore::KoResourceTagStore(KoResourceServerBase *resourceServer)
 
 KoResourceTagStore::~KoResourceTagStore()
 {
-    serializeTags();
     delete d;
 }
 
@@ -288,7 +290,6 @@ void KoResourceTagStore::writeXMLFile(const QString &tagstore)
             root.appendChild(resourceEl);
         }
     }
-
     QTextStream metastream(&f);
     metastream.setCodec("UTF-8");
     metastream << doc.toString();
-- 
2.24.0

